packages/cli/src/daemon/error-handler.ts

```diff
@@ -0,0 +1,20 @@
+import { ErrorRequestHandler, NextFunction, Request, Response } from 'express';
+import { DiagnosticsLogger } from '@ceramicnetwork/common';
+
+/**
+ * Generic error handling middleware for the daemon.
+ */
+export function errorHandler(logger: DiagnosticsLogger): ErrorRequestHandler {
+  return (err: Error, req: Request, res: Response, next: NextFunction) => {
+    (req as any).error = err // Allow other middlewares access the error
+    logger.err(err);
+    if (res.headersSent) {
+      return next(err);
```

Comment:
Does this mean express will handle the error?

Reply:
"This" was in reference to `next(err)`

Reply:
> If you call next() with an error after you have started writing the response (for example, if you encounter an error while streaming the response to the client) the Express default error handler closes the connection and fails the request.

> So when you add a custom error handler, you must delegate to the default Express error handler, when the headers have already been sent to the client

https://expressjs.com/en/guide/error-handling.html

Reply:
TLDR It will close the connection and fail the request. Most probably never happens.

Reply:
Aha thanks